packages/curve-ui-kit/src/shared/ui/FormTabs/FormTabs.tsx

```diff
@@ -0,0 +1,76 @@
+import { type ReactNode, useState } from 'react'
+import Stack from '@mui/material/Stack'
+import { AppFormContentWrapper } from '@ui/AppForm'
+import { type TabOption, TabsSwitcher } from '@ui-kit/shared/ui/TabsSwitcher'
+import { WithWrapper } from '@ui-kit/shared/ui/WithWrapper'
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+
+const { MaxWidth } = SizesAndSpaces
+
+export type FormTab<Props> = {
+  value: string
+  label: string | ((props: Props) => string)
+  subTabs?: Pick<FormTab<Props>, 'value' | 'label' | 'component'>[]
+  enabled?: (props: Props) => boolean | undefined
```

Comment:
I wouldn't call this `enabled`, but rather `visible`. Or keep both if needed. The `TabSwitcher` component namely can disable tabs while they still remain visible.